homework/src/index.js

```diff
@@ -1,47 +1,150 @@
 'use strict';
 
-{
-  function fetchJSON(url, cb) {
-    const xhr = new XMLHttpRequest();
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status < 400) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
-  }
-
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach((key) => {
-      const value = options[key];
-      if (key === 'html') {
-        elem.innerHTML = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
-    });
-    return elem;
-  }
-
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
-    });
-  }
-
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-
-  window.onload = () => main(HYF_REPOS_URL);
+function main() {
+    const url = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+
+    function createAndAppend(tagName, parent) {
+        const element = document.createElement(tagName);
+        parent.appendChild(element);
+        return element;
+    }
+
+    function fetchJSON(url, cb) {
+        const xhr = new XMLHttpRequest();
+        xhr.open('GET', url, true);
+        xhr.responseType = 'json';
+        xhr.onreadystatechange = () => {
+            if (xhr.readyState === 4) {
+                if (xhr.status < 400) {
+                    cb(null, xhr.response);
+                } else {
+                    cb(new Error(`Network error : ${xhr.status} - ${xhr.statusText}`));
+                }
+            }
+        };
+        xhr.send();
+    }
+
+    function callBack(error, data) {
+        if (error !== null) {
+            const err = document.getElementById('root');
+            err.innerHTML = error.message;
+        } else {
+            renderRepositories(null, data);
+        }
+    }
+
+    function renderRepositories(error, data) {
+        if (error !== null) {
+            const err = document.getElementById('root');
+            err.innerHTML = error.message;
```

Comment:
You are handling errors in two places, should handle only at one place to avoid complexity